apps/asn1pars.c

```diff
@@ -226,7 +226,7 @@ int asn1parse_main(int argc, char **argv)
             ASN1_TYPE *atmp;
             int typ;
             j = atoi(sk_OPENSSL_STRING_value(osk, i));
-            if (j == 0) {
+            if (j <= 0 || j >= tmplen) {
                 BIO_printf(bio_err, "'%s' is an invalid number\n",
```

Comment:
It's not your code, but I would prefer the message "'%s' is out of range\n" instead.

Reply:
Hm... on the other hand, j == 0 can indeed mean that the input is not a valid number. To distinguish the cases "not a valid number" and "number out of range", one would have to use `strtoul` instead of `atoi`.  So, never mind, just forget my last comment.

Reply:
Look at using opt_int()

Reply:
I could use 'p' in  `{"strparse", OPT_STRPARSE, 'p'` then the number format is checked
by the opt machinery, and print x is out of range here.
How does that sound?

Reply:
sounds good.  approval to do that.